### PR TITLE
Docs/fix documentation generation

### PR DESCRIPTION
**Pull request by monitz87**
### Description

- Upgrade typedoc and the markdown plugin to the latest versions
- Migrate the typedoc config to be compatible
- Refactor the `Polymesh` class and its namespaces into `api/client`
- Hide some types that shouldn't be exposed by the docs

These changes keep the doc generation from breaking and result in something acceptable, but not ideal. I created [this](https://polymath.atlassian.net/browse/DA-350) task to discuss improvements

### JIRA Link

https://polymath.atlassian.net/browse/DA-347

### Checklist

- [x] Updated the Readme.md (if required) ?

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -30,5 +30,5 @@
     "lib": ["es2017", "dom"],
     "typeRoots": ["./node_modules/@types", "./src/typings"]
   },
-  "exclude": ["dist", "node_modules", "sandbox.ts", "test.ts"]
+  "exclude": ["dist", "node_modules", "sandbox.ts", "test.ts", "src-temp"]
```
Comment (polymath-eric) [inline tsconfig.json:33]: Do you still need `src-temp` here?

File: src/api/client/__tests__/Settlements.ts [hunk 2]
```diff
@@ -1,12 +1,13 @@
 import BigNumber from 'bignumber.js';
 
 import { addInstructionTransformer, Context, TransactionQueue, Venue } from '~/internal';
-import { Settlements } from '~/Settlements';
 import { dsMockUtils, entityMockUtils, procedureMockUtils } from '~/testUtils/mocks';
 import { Mocked } from '~/testUtils/types';
 import { Instruction, VenueType } from '~/types';
 import { InstructionAffirmationOperation } from '~/types/internal';
 
+import { Settlements } from '../Settlements';
```
Comment (polymath-eric) [inline src/api/client/__tests__/Settlements.ts:9]: (nit) I thought we mostly used `~` as convention.

I am not against relative paths, but I like the consistency of `~` everywhere.

**Review by polymath-eric — COMMENTED** [review]
Seems OK to me. 

I think `/client` is a good name for the namespace stuff.

File: tsconfig.json (hunk 1 above)
Comment (monitz87) [reply tsconfig.json:33]: Yeah. The new typedoc version just tries to build everything it finds

File: src/api/client/__tests__/Settlements.ts (hunk 2 above)
Comment (monitz87) [reply src/api/client/__tests__/Settlements.ts:9]: Fair enough, I'll fix it

**Review by VictorVicente — APPROVED** [review]

**Review by polymath-eric — APPROVED** [review]

**Pull request merged**